services/shhext/chat/multidevice/sql_lite_persistence.go

```diff
@@ -52,20 +55,116 @@ func (s *SQLLitePersistence) GetActiveInstallations(maxInstallations int, identi
 
 }
 
+// GetInstallations returns all the installations for a given identity
+// we both return the installations & the metadata
+// metadata is currently stored in a separate table, as in some cases we
+// might have metadata for a device, but no other information on the device
+func (s *SQLLitePersistence) GetInstallations(identity []byte) ([]*Installation, error) {
+	installationMap := make(map[string]*Installation)
+	var installations []*Installation
+
+	// We query both tables as sqlite does not support full outer joins
+	installationsStmt, err := s.db.Prepare(`SELECT installation_id, version, enabled, timestamp FROM installations WHERE identity = ?`)
+	if err != nil {
+		return nil, err
+	}
+	defer installationsStmt.Close()
+
+	installationRows, err := installationsStmt.Query(identity)
+	if err != nil {
+		return nil, err
+	}
+
+	for installationRows.Next() {
+		var (
+			installationID string
+			version        uint32
+			enabled        bool
+			timestamp      int64
+		)
+		err = installationRows.Scan(
+			&installationID,
+			&version,
+			&enabled,
+			&timestamp,
+		)
+		if err != nil {
+			return nil, err
+		}
+		installationMap[installationID] = &Installation{
+			ID:                   installationID,
+			Version:              version,
+			Enabled:              enabled,
+			Timestamp:            timestamp,
+			InstallationMetadata: &InstallationMetadata{},
```

Comment:
Only in this endpoint, other endpoint don't return the metadata as it's not needed, so unless you feel strongly I would keep it as a pointer

Reply:
No, if neither case covers 100%, let's keep it. I would just add a comment in this line why we do it this way as it's a bit odd and in a normal situation, it would be skipped.